### PR TITLE
ConvBERT fix torch <> tf weights conversion

### PR DESCRIPTION
**Pull request by abhishekkrthakur**
(from @patrickvonplaten):

This PR corrects the shape of the grouped linear layer weight so that the general conversion function does not have to be changed.

All models are tested to work correctly as follows:

```python
from transformers import ConvBertModel, TFConvBertModel
import tensorflow as tf
import torch

input_ids = [[1, 2, 3, 4, 5]]

tf_input_ids = tf.convert_to_tensor(input_ids)
pt_input_ids = torch.tensor(input_ids)

for name in ["conv-bert-base", "conv-bert-medium-small", "conv-bert-small"]:
    model_tf = TFConvBertModel.from_pretrained(f"YituTech/{name}", from_pt=True)
    model = ConvBertModel.from_pretrained(f"YituTech/{name}")

    assert abs(model_tf(tf_input_ids)[0].cpu().numpy().sum() - model(pt_input_ids)[0].cpu().numpy().sum()) < 1e-2, "Error"
```

Changing the size and the name of a weight means that all tf weights have to be updated, but I think this is ok here since the TF models (if I understood correctly) were not behaving as expected before anyways.

I also checked that the conversion the other way around works as expected (`...from_tf=True`)

### CODE REVIEW

File: src/transformers/modeling_tf_pytorch_utils.py [hunk 1]
```diff
@@ -178,6 +184,24 @@ def load_pytorch_weights_in_tf2_model(tf_model, pt_state_dict, tf_inputs=None, a
...
+
+        if name.endswith("conv_attn_key.bias"):
+            array = numpy.squeeze(array)
+            transpose = False
+
+        if 1 == 0:
+            if name.endswith("intermediate.dense.weight") or name.endswith("output.dense.weight"):
+                if not name.endswith("attention.output.dense.weight"):
+                    print(name)
+                    transpose = False
```
Comment (jplu) [inline src/transformers/modeling_tf_pytorch_utils.py:203]: Does `medium` or `small` appears in the name? If yes we can use the full name of the model to activate/deactivate this branch.

Also I don't get this double `if` condition, why not simply use the second `if`? Does the `attention.output.dense` layer appears only in the `small` or `medium` version of the model?

**Review by jplu — COMMENTED** [review]
In overall that's ok, still need this weird condition part to better handle.

File: tests/test_modeling_tf_convbert.py [hunk 2]
```diff
@@ -392,21 +392,19 @@ def check_encoder_attentions_output(outputs):
 class TFConvBertModelIntegrationTest(unittest.TestCase):
     @slow
     def test_inference_masked_lm(self):
-        model = TFConvBertModel.from_pretrained("YituTech/conv-bert-base")
+        model = TFConvBertModel.from_pretrained("YituTech/conv-bert-base", from_pt=True)
```
Comment (patrickvonplaten) [inline tests/test_modeling_tf_convbert.py:395]: This needs to be changed once the TF weights are correctly uploaded

File: src/transformers/models/convbert/modeling_tf_convbert.py [hunk 3]
```diff
@@ -343,7 +343,7 @@ def __init__(self, input_size, output_size, num_groups, kernel_initializer, **kw
     def build(self, input_shape):
         self.kernel = self.add_weight(
             "kernel",
-            shape=[self.num_groups, self.group_in_dim, self.group_out_dim],
+            shape=[self.group_out_dim, self.group_in_dim, self.num_groups],
```
Comment (patrickvonplaten) [inline src/transformers/models/convbert/modeling_tf_convbert.py:346]: this allows us to keep the conversion function general

**Review by sgugger — APPROVED** [review]
Very clean! Thanks a lot for fixing the issue!

**Review by LysandreJik — APPROVED** [review]
Ok with the change!

Should we do a patch release for this?

**Comment by abhishekkrthakur** [comment]
I'll push the fixed weights and remove from_pt in test.

**Comment by patrickvonplaten** [comment]
> Ok with the change!
> 
> Should we do a patch release for this?

Think it's a good idea

**Comment by abhishekkrthakur** [comment]
Thanks @patrickvonplaten

**Comment by LysandreJik** [comment]
Ok will do a patch this afternoon

**Comment by LysandreJik** [comment]
cc @stefan-it and @mrm8488 that have been playing with the model. We'll release v4.3.3 very soon which will contain that patch.

**Comment by stefan-it** [comment]
Should I re-convert the TF model then :thinking:

**Comment by LysandreJik** [comment]
I think you should, once v4.3.3 is released (in a few minutes)

**Comment by LysandreJik** [comment]
v4.3.3 has been released!

**Pull request merged**